### PR TITLE
feat: added filter by publication metadata

### PR DESCRIPTION
**Pull request by MillenniumFalconMechanic**
### Reviewers
**Functional:** 
@tihuan 

---

## Changes
- Added filter by publication date to collections and datasets.
- Added filter by publication author to collections and datasets.

### CODE REVIEW

File: frontend/src/common/hooks/useCategoryFilter.ts [hunk 1]
```diff
@@ -166,6 +168,29 @@ function addEmptyCategoryValues(
   }
 }
 
+/**
```
Comment (MillenniumFalconMechanic) [inline frontend/src/common/hooks/useCategoryFilter.ts:171]: I added this function as a way to swap out the filter type per column. I initially anticipated using a `between` filter type for filter by publication date but ended up using the existing `includesSome`. I have kept the functionality as-is (that is, always using `includesSome`) but I expect we will need it - with modifications - for future filters such as cell count, gene count and possible development stage.

File: frontend/src/common/entities.ts [hunk 2]
```diff
@@ -58,6 +58,15 @@ export enum IS_PRIMARY_DATA {
   BOTH = "BOTH",
 }
 
+/**
+ * Author of publication associated with a collection, populated from Crossref as part of collection publication
+ * metadata.
+ */
+export interface Author {
+  family: string;
```
Comment (tihuan) [inline frontend/src/common/entities.ts:66]: ooh I like this over first name last name, since different countries do first name last name differently! 💡

**Review by tihuan — APPROVED** [review]
Wow super amazing PR 😮 👏 Love your approach for filtering date by indexing it upfront 😆  Thanks so much for adding all the comments and tests too 🙏

**Comment by MillenniumFalconMechanic** [comment]
Thank you for taking the time to review this PR @tihuan!

**Pull request merged**